Observer/SourceDeductionProcessor.php

```diff
@@ -87,6 +111,42 @@ public function execute(EventObserver $observer)
 
         $sourceSelectionResult = $this->getSourceSelectionResultFromOrder->execute($order);
 
+        if (!$extensionAttributes = $order->getExtensionAttributes()) {
+            $extensionAttributes = $this->orderExtensionFactory->create();
+        }
+
+        $extensionAttributes->setSources(
+            $sourcesItems = $sourceSelectionResult->getSourceSelectionItems()
+        );
+
+        $order->setExtensionAttributes($extensionAttributes);
+
+        $sources = [];
+        foreach ($sourcesItems as $item) {
+            $sources[] = [
+                'source_code' => $item->getSourceCode(),
```

Comment:
I don't think we should be doing it this way, we can probably add sources in the same way we were doing before and have the BE serialise the models when saving. Can we maybe abstract this away into a different class/repository that handles saving/loading of these extension attributes?